generate_uninstall/CMakeLists.txt

```diff
@@ -3,6 +3,14 @@
 
 set (UNINSTALL_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 
+# When added as a subdirectory in the mongocxx driver, there is a conflict in
+# the `uninstall` target causing the `make uninstall` command to fail. By
+# defining the uninstall target as a variable, the C++ driver is able to rename
+# this target in order to avoid the conflict.
+if (NOT DEFINED MONGOC_UNINSTALL_TARGET)
+   set(MONGOC_UNINSTALL_TARGET uninstall)
```

Comment:
If the C++ driver renames the target, will running the uninstall target in the C++ driver only uninstall the C++ driver (and leave the C driver installed files)?

Reply:
In this PR, `make uninstall` will remove both the C and C++ drivers.

```
$ cmake .. -DCMAKE_INSTALL_PREFIX=/tmp/cxxdriver
$ make -j16 && make install
$ tree -L 2 /tmp/cxxdriver/
/tmp/cxxdriver/
├── include
│   ├── bsoncxx
│   └── mongocxx
├── lib
│   ├── cmake
│   ├── libbsoncxx.so -> libbsoncxx.so._noabi
│   ├── libbsoncxx.so.0.0.0
│   ├── libbsoncxx.so._noabi -> libbsoncxx.so.0.0.0
│   ├── libmongocxx.so -> libmongocxx.so._noabi
│   ├── libmongocxx.so.0.0.0
│   ├── libmongocxx.so._noabi -> libmongocxx.so.0.0.0
│   └── pkgconfig
└── share
    └── mongo-cxx-driver

8 directories, 6 files
$ tree -L 2 /tmp/cxxdriver/
/tmp/cxxdriver/ [error opening dir]

0 directories, 0 files
```

Reply:
With this PR, `make uninstall` will uninstall both the C and C++ drivers:

```
$ make install
$ tree -L 2 /tmp/cxxdriver/
/tmp/cxxdriver/
├── bin
│   └── mongoc-stat
├── include
│   ├── bsoncxx
│   ├── libbson-1.0
│   ├── libmongoc-1.0
│   └── mongocxx
├── lib
│   ├── cmake
│   ├── libbson-1.0.so -> libbson-1.0.so.0
│   ├── libbson-1.0.so.0 -> libbson-1.0.so.0.0.0
│   ├── libbson-1.0.so.0.0.0
│   ├── libbson-static-1.0.a
│   ├── libbsoncxx.so -> libbsoncxx.so._noabi
│   ├── libbsoncxx.so.0.0.0
│   ├── libbsoncxx.so._noabi -> libbsoncxx.so.0.0.0
│   ├── libmongoc-1.0.so -> libmongoc-1.0.so.0
│   ├── libmongoc-1.0.so.0 -> libmongoc-1.0.so.0.0.0
│   ├── libmongoc-1.0.so.0.0.0
│   ├── libmongoc-static-1.0.a
│   ├── libmongocxx.so -> libmongocxx.so._noabi
│   ├── libmongocxx.so.0.0.0
│   ├── libmongocxx.so._noabi -> libmongocxx.so.0.0.0
│   └── pkgconfig
└── share
    ├── mongo-c-driver
    └── mongo-cxx-driver

12 directories, 15 files
$ make uninstall
$ tree -L 2 /tmp/cxxdriver/
/tmp/cxxdriver/ [error opening dir]

0 directories, 0 files
```